langchain/document_loaders/text.py

```diff
@@ -1,20 +1,55 @@
-from typing import List, Optional
+from typing import TYPE_CHECKING, List, Optional
+
+import chardet
```

Comment:
let's make this optional (so only import within function that it's used and add to pyproject.toml as optional